lib/pharos/phases/configure_client.rb

```diff
@@ -6,17 +6,36 @@ class ConfigureClient < Pharos::Phase
       title "Configure kube client"
 
       def call
-        Dir.mkdir(config_dir, 0o700) unless Dir.exist?(config_dir)
-        config_file = File.join(config_dir, @host.api_address)
+        install_local_kubeconfig(read_kubeconfig)
+        install_remote_kubeconfig
+      end
+
+      def config_dir
+        File.join(Dir.home, '.pharos')
+      end
+
+      def read_kubeconfig
         logger.info { "Fetching kubectl config ..." }
         config_data = @ssh.file("/etc/kubernetes/admin.conf").read
+        config_data = config_data.gsub(%r{(server: https://)(.+)(:6443)}, "\\1#{@host.api_address}\\3")
+      end
+
+      def install_local_kubeconfig(config_data)
+        Dir.mkdir(config_dir, 0o700) unless Dir.exist?(config_dir)
+
+        config_file = File.join(config_dir, @host.api_address)
```

Comment:
This should be `@host.address`, I think...

Reply:
Why?

Reply:
I suppose if we're using the `api.endpoint` from the config for the CLI => Master requests, then we can't run the `ConfigureClient` phase for all master hosts... need to move the remote kubeconfig thing back out of this phase.

Reply:
See #278 that changes this to use `@config.api_endpoint` instead.